src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs

```diff
@@ -172,6 +208,22 @@ public IReadOnlyList<DataColumn> GetAllUsedDataColumns()
          return _dataColumnDTOCache.KeyValues.Where(x => x.Value.Used).Select(x => x.Key).ToList();
       }
 
+      public OutputMappings AllOutputMappings { get; set; }
+
+      public void OutputObservedDataLinkingChanged(bool isLinkedMappedOutputs)
+      {
+         _isLinkedMappedOutputs = isLinkedMappedOutputs;
+
+         if (!_isLinkedMappedOutputs) return;
+
+         foreach (var dataColumnPair in _dataColumnDTOCache.KeyValues)
+         {
+            var outputColumnUsed = dataColumnPair.Value.Used;
+            var linkedObservedData = getLinkedObservedDataFromOutputPath(dataColumnPair.Key.PathAsString);
```

Comment:
This will also iterate over observed data column no? 
You can iterate over the _dataColumnDTOCache and access the Path or the datacolumn directly. No need to use the internal of the cache here

Reply:
in fact getLinkedObservedDataFromOutputPath could simply given a DataColumn as Parameter the way it's being used

Reply:
Hmm not sure how this would work... we need the "Used" value from DataColumnDTO and the "PathAsString" from the corresponding DataColumn...

Reply:
done